src/core/mount.c

```diff
@@ -2106,11 +2124,27 @@ static void mount_enumerate(Manager *m) {
                 }
 
                 (void) sd_event_source_set_description(m->mount_event_source, "mount-monitor-dispatch");
+
+                if (!MANAGER_IS_SYSTEM(m)) {
+                        r = sd_event_add_post(m->event, &m->mount_enumerate_source, mount_enumerate_late, m);
+                        if (r < 0) {
+                                log_error_errno(r, "Failed to add late mount enumerate source: %m");
+                                goto fail;
+                        }
+
+                        r = sd_event_source_set_priority(m->mount_enumerate_source, SD_EVENT_PRIORITY_IDLE);
+                        if (r < 0) {
+                                log_error_errno(r, "Failed to adjust mount enumerate priority: %m");
+                                goto fail;
+                        }
+                }
         }
 
-        r = mount_load_proc_self_mountinfo(m, false);
-        if (r < 0)
-                goto fail;
+        if (MANAGER_IS_SYSTEM(m)) {
```

Comment:
this conditionalization sounds really weird to me. if this is safe to do, why wouldn't this be safe for the system too?